chain/client/src/stateless_validation/chunk_validator/mod.rs

```diff
@@ -650,13 +650,38 @@ impl Client {
             }
             Err(err) => return Err(err),
         };
+
+        // Send the acknowledgement for the state witness back to the chunk producer.
+        // This is currently used for network roundtrip time measurement, so we do not need to
+        // wait for validation to finish.
+        if let Err(err) = self.send_state_witness_ack(&witness) {
```

Comment:
Let's move this logic to the beginning of the `process_chunk_state_witness` function. We want to send the ack even if the witness is added to the orphan pool, but currently that won't happen because of early returns.